### PR TITLE
Adds password strength validation to users api

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Any passwords passed to createUser or updateUser will now be
validated to be at least 8 characters long and sufficiently
complex.

medic/medic-webapp#1472

# Review checklist

- [x] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [x] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [x] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [x] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -402,6 +401,15 @@ var deleteUser = function(id, callback) {
   });
 };
 
+const validatePassword = password => {
+  if (password.length < PASSWORD_MINIMUM_LENGTH) {
+    return error400(`The password must be at least ${PASSWORD_MINIMUM_LENGTH} characters long.`);
```
Comment (SCdF) [inline controllers/users.js]: Who is consuming these error messages? Do they need to be internationalised?

**Comment by SCdF** [comment]
@garethbowen not sure if i18n is required here. Depends on consumer.

File: controllers/users.js (hunk 1 above)
Comment (garethbowen) [inline controllers/users.js]: I don't think so and none of them are. These aren't shown in the webapp anywhere just in the command line.

File: controllers/users.js [hunk 2]
```diff
@@ -488,32 +494,33 @@ module.exports = {
...
-        series = [],
-        response = {},
-        settings,
-        user;
-    var props = _.uniq(USER_EDITABLE_FIELDS.concat(SETTINGS_EDITABLE_FIELDS));
-    var hasFields = _.some(props, function(k) {
+    const self = this,
+          userID = createID(username);
+    const props = _.uniq(USER_EDITABLE_FIELDS.concat(SETTINGS_EDITABLE_FIELDS));
+    const hasFields = _.some(props, function(k) {
```
Comment (garethbowen) [inline controllers/users.js]: :(

**Pull request merged**